packages/material-ui/src/Popover/Popover.d.ts

```diff
@@ -131,6 +131,11 @@ export interface PopoverProps
    * @default 'auto'
    */
   transitionDuration?: TransitionProps['timeout'] | 'auto';
+  /**
+   * Set to 'true' to put the transition component inside the paper element. This is necessary
+   * for transition components that change the height of the popover (like Collapse).
```

Comment:
Can we instead make this the new default behavior and don't branch at all? That way we don't have to teach another thing and reduce the potential of author-error.

Reply:
Good point. Ya, I was worried that there might be use-cases where people wanted the Paper element to transition, but it's probably better to start simple and cross that bridge when we get to it. Changed!

Reply:
I wouldn't be surprised if we do need to support both cases. Ideally we'd have an implementation that doesn't require extra author attention because it's yet another thing they need to be aware of.

I'm less concerned about changing it now since the change would be released in an alpha channel. Let's what other maintainers think.

Reply:
unfortunately, this approach won't work : (

Here's the problem:
- The Popover Component accepts multiple children
- Some transition components (like Grow) require a single content child https://material-ui.com/api/grow/
So this layout won't work in that case

I reverted it back to a transitionInsidePaper parameter. open to other ideas

Reply:
(for posterity, here's everything that broke when i tried to change the default: https://app.circleci.com/pipelines/github/mui-org/material-ui/31546/workflows/559b2e5d-00fc-4048-9e5a-8ad7f0728f4c/jobs/209919)